packages/spark-core/tests/modals.tests.js

```diff
@@ -263,6 +478,120 @@ describe('Modal tests', () => {
+      key: 'Q',
+      keyCode: 81,
+      preventDefault: () => {},
+    };
+    showModal(waitModal, modalMask, main);
+    handleModalKeyEvents(modalsList, modalMask, main, QKeyEvent);
+    expect(document.activeElement).eql(waitModal);
+  });
+
+  it('should return focus to the open wait modal if tab is pressed', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
+      keyCode: 9,
+      preventDefault: () => {},
+    };
+    showModal(waitModal, modalMask, main);
+    handleModalKeyEvents(modalsList, modalMask, main, tabKeyEvent);
+    expect(document.activeElement).eql(waitModal);
+  });
+
+  it('should return focus to the open wait modal if shift + tab is pressed', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
+      keyCode: 9,
+      shiftKey: true,
+      preventDefault: () => {},
+    };
+    showModal(waitModal, modalMask, main);
+    handleModalKeyEvents(modalsList, modalMask, main, tabKeyEvent);
+    expect(document.activeElement).eql(waitModal);
+  });
+
+  it('should return focus to the first element if tab is pressed while the last focusable element is focused', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
+      keyCode: 9,
+      preventDefault: () => {},
+    };
+    showModal(defaultModal, modalMask, main);
+    link2.focus();
+    expect(document.activeElement).eql(link2);
+    handleModalKeyEvents(modalsList, modalMask, main, tabKeyEvent);
+    expect(document.activeElement).eql(cancelDefault);
+  });
+
+  it('should return focus to the first element if tab + shift is pressed while the first focusable element is focused', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
+      keyCode: 9,
+      shiftKey: true,
+      preventDefault: () => {},
+    };
+    showModal(defaultModal, modalMask, main);
+    cancelDefault.focus();
+    expect(document.activeElement).eql(cancelDefault);
+    handleModalKeyEvents(modalsList, modalMask, main, tabKeyEvent);
+    expect(document.activeElement).eql(link2);
+  });
+
+  it('should do nothing if tab + shift is pressed while the focused element is not first', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
+      keyCode: 9,
+      shiftKey: true,
+      preventDefault: () => {},
+    };
+    showModal(defaultModal, modalMask, main);
+    link2.focus();
+    handleModalKeyEvents(modalsList, modalMask, main, tabKeyEvent);
+    expect(document.activeElement).eql(link2);
+  });
+
+  it('should do nothing if tab is pressed while the focused element is not last', () => {
+    const modalsList = document.querySelectorAll('[data-sprk-modal]');
+    const tabKeyEvent = {
+      key: 'Tab',
```

Comment:
now that we are using sinon - maybe these event vars should be switched to the new way